pmd-core/src/main/java/net/sourceforge/pmd/cache/AnalysisResult.java

```diff
@@ -49,7 +49,10 @@ private static long computeFileChecksum(final File sourceFile) {
             // the analysis will fail and report the error on it's own since the checksum won't match
         }
 
-        return 0;
+        // we couldn't read the file, maybe the file doesn't exist
+        // in any case, we can't use the cache. Returning here the timestamp should make
+        // sure, we see that the file changed every time we analyze it.
+        return System.currentTimeMillis();
```

Comment:
this seems unrelated to the rest of the PR, although related to the original report

Reply:
Yes, this would be fixing the symptom, so that the cache would be essentially disabled when using shortnames... I can separate it out in a extra PR.

Reply:
Here it is: #2000